test/bigint_test.dart

```diff
@@ -31,6 +31,16 @@ main() {
       expect(p('-0$n').toString(), equals('-$n'));
     }
   });
+  test('toDouble()', () {
+    [ '0', '1', '-1', '2', '-2', '23', '-23', '-0',
+      '3187801890382', '-3187801890382',
+    ].forEach((n) =>
+        expect(p(n).toDouble().toString().replaceAll(".0", ""), n));
+    expect(p('31878018903828899277492024491376690701584023926880').toDouble().toString(),
+        allOf([startsWith('3.187801890382'), endsWith('e+49')]));
+    expect(p('-31878018903828899277492024491376690701584023926880').toDouble().toString(),
+        allOf([startsWith('-3.187801890382'), endsWith('e+49')]));
+  });
```

Comment:
Can you use `double.parse` instead of String comparaison ?
Something like this : 

```dart
  test('toDouble()', () {
    [
      '0',
      '1',
      '2',
      '23',
      '3187801890382',
      '31878018903828899277492024491376690701584023926880',
    ]
        .expand((e) => [e, '-$e']) // also test negative values
        .forEach((n) => expect(p(n).toDouble(), equals(double.parse(n))));
  });
```